android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/RunFragment.kt

```diff
@@ -25,17 +26,24 @@ class RunFragment : Fragment() {
     // TODO: Rename and change types of parameters
     private var param1: String? = null
     private var param2: String? = null
+
+    private lateinit var stopwatchViewModel: StopwatchView
     private var pauseOffset = 0L
     private var isWorking = false
+    private var fragmentPauseStartTime = 0L
 
     private fun startStopTimer(runBtn: Button, chronometer : Chronometer) {
         isWorking = if (!isWorking) {
             chronometer.base = SystemClock.elapsedRealtime() - pauseOffset
             chronometer.start()
+            chronometer.showContextMenu()
+            stopwatchViewModel.setWorkingStatus(true)
             true
         } else {
             pauseOffset = SystemClock.elapsedRealtime() - chronometer.base
             chronometer.stop()
+            stopwatchViewModel.setPauseOffset(SystemClock.elapsedRealtime() - chronometer.base)
+            stopwatchViewModel.setWorkingStatus(false)
             false
         }
         runBtn.setText(if (!isWorking) R.string.start else R.string.stop)
```

Comment:
Since we are using the ViewModel, this can be done with data binding. Something like this in your layout
```
<Button
    android:id="@+id/xxx"

    ...
    android:text="@{viewModel.isWorking ? @string/stop : @string/start}"
    android:onClick="@{() -> viewModel.updateIsWorking()}"
>
```

More info https://developer.android.com/topic/libraries/data-binding